c10/core/TensorAxes.cpp

```diff
@@ -0,0 +1,23 @@
+#include <c10/core/TensorAxes.h>
+
+#include <iostream>
+
+namespace c10 {
+
+std::string TensorAxes::toString() const {
+  std::stringstream ss;
+  ss << *this;
+  return ss.str();
```

Comment:
It may not matter much, but I think this is introducing an extra string copy.  The `stringstream` will have to copy from its internal storage into a temporary string.  It would be more efficient to construct a string first, pass it to the by-reference constructor to `stringstream`, and then return it at the end.